src/main/java/com/shc/silenceengine/graphics/cameras/FPSCamera.java

```diff
@@ -139,13 +140,20 @@ public FPSCamera moveDown(float amount)
 
     public FPSCamera rotateX(float angle)
     {
+        // If the current angle is already on the limit, do nothing.
+        if (angleX <= -ANGLE_LIMIT_X || angleX >= ANGLE_LIMIT_X) {
```

Comment:
If we are already inside the limit of the rotation, we must not rotate it's like this part of your code :
```
if (angleX < -ANGLE_LIMIT_X || angleX > ANGLE_LIMIT_X)
{
    angleX -= angle;
    return this;
}
```